EIPS/eip-draft_transaction_data_opcodes.md

```diff
@@ -0,0 +1,106 @@
+---
+eip: <to be assigned>
+title: Transaction Data Opcodes
+author: Alex Papageorgiou (@alex-ppg)
+discussions-to: https://ethereum-magicians.org/t/eip-draft-transaction-data-opcodes/6017
+status: Draft
+type: Standards Track
+category: Core
+created: 2021-04-16
+---
+
+## Simple Summary
+
+Provide access to original transaction data.
+
+## Abstract
+
+This EIP introduces the following four EVM instructions: `ORIGINDATALOAD`, `ORIGINDATASIZE`, `ORIGINDATACOPY` and `ENTRYPOINT`.
+
+The first three instructions are meant to provide access to the original transaction's `data` payload whereas the last instruction is meant to provide access to the original recipient of the transaction, the `to` address.
+
+## Motivation
+
+It is undeniable that smart contracts are becoming more interconnected than ever. Up until this point, smart contracts have entirely relied on compliant interfaces and introspection to introduce a new step in the call chain of a complex multi-contract interaction. However, this presents a forwards-only approach which limits the types of interactions that can manifest.
+
+The purpose of this EIP is to provide a way via which a contract is able to identify the entry-point of a transaction on the blockchain and deduce what was the original intention of the transaction by applying introspection on the original transaction data itself.
+
+This EIP enables the development of new types of smart contracts as it can open new pathways for [EIP-721](https://eips.ethereum.org/EIPS/eip-721) NFTs and [EIP-20](https://eips.ethereum.org/EIPS/eip-20) tokens to detect which action their transaction is part of, such as detecting a liquidity provision to a decentralized exchange or a loan within a collateralized lending protocol.
```

Comment:
```suggestion
This EIP enables the development of new types of smart contracts as it can open new pathways for [EIP-721](./eip-721.md) NFTs and [EIP-20](./eip-20.md) tokens to detect which action their transaction is part of, such as detecting a liquidity provision to a decentralized exchange or a loan within a collateralized lending protocol.
```